### PR TITLE
Uninstall module used on PS 1.6 before using this one

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
## Table changes between this module & blocklayered:

* New DB table: layered_filter_block
* Removed DB Table: layered_friendly_url

## DB Changes:
* layered_filter
  * filters => LONGTEXT
* layered_price_index
  * + id_country
  * index to replace
* layered_category
  * charset latin1 -> utf8
 

This change is [Reviewable](https://reviewable.io/reviews/prestashop/ps_facetedsearch/94)

### CODE REVIEW

File: ps_facetedsearch.php [hunk 1]
```diff
@@ -174,31 +179,43 @@ public function install()
...
-        Configuration::updateValue('PS_LAYERED_FILTER_PRICE_ROUNDING', 1);
-
-        $this->psLayeredFullTree = 1;
-
-        $this->rebuildLayeredStructure();
-        $this->buildLayeredCategories();
-
-        $productsCount = $this->getDatabase()->getValue('SELECT COUNT(*) FROM `' . _DB_PREFIX_ . 'product`');
+        if ($this->uninstallPrestaShop16Module()) {
+            // ToDo: Does `layered_category` stores critical data?
```
Comment (Quetzacoalt91) [inline ps_facetedsearch.php]: Check me

File: ps_facetedsearch.php [hunk 2]
```diff
@@ -174,31 +179,43 @@ public function install()
...
+            $this->rebuildLayeredStructure();
 
-        if ($productsCount < static::LOCK_TEMPLATE_CREATION) {
-            $this->rebuildLayeredCache();
-        }
-
-        $this->rebuildPriceIndexTable();
-        $this->installIndexableAttributeTable();
-        $this->installProductAttributeTable();
+            // ToDo: Does `layered_price_index` stores critical data ? Probably not with `_index`
```
Comment (Quetzacoalt91) [inline ps_facetedsearch.php]: Check me

File: ps_facetedsearch.php [hunk 3]
```diff
@@ -174,31 +179,43 @@ public function install()
...
+            $this->rebuildPriceIndexTable();
 
-        if ($productsCount < static::LOCK_TOO_MANY_PRODUCTS) {
-            $this->fullPricesIndexProcess();
-            $this->indexAttributes();
+            $this->getDatabase()->execute('TRUNCATE TABLE ' . _DB_PREFIX_ . 'layered_filter CHANGE `filters` `filters` LONGTEXT NULL');
+            $this->getDatabase()->execute('DROP TABLE IF EXISTS ' . _DB_PREFIX_ . 'friendly_url');
+            // ToDo: What cache do we need to (re)generate ?
```
Comment (Quetzacoalt91) [inline ps_facetedsearch.php]: Cache generation needed here?

Comment (PierreRambaud) [reply ps_facetedsearch.php]: We mustn't generate cache while upgrading, it takes too many times (really too many)

**Comment by PierreRambaud** [comment]
Thanks @Quetzacoalt91

**Pull request merged**